lib/matplotlib/artist.py

```diff
@@ -93,6 +100,50 @@ class Artist:
 
     zorder = 0
 
+    def __init_subclass__(cls):
+        # Inject custom set() methods into the subclass with signature and
+        # docstring based on the subclasses' properties.
+        # docstring based on the subclasses' properties.
+
+        if 'set' in cls.__dict__:
+            return  # don't overwrite set if the subclass defines set itself.
```

Comment:
I think it's more `if cls.set != Artist.set` (because cls could also be a subclass of a subclass that overrides `set`)

Reply:
`if cls.set != Artist.set: return` does not work:

Consider the class hierarchy `Artist` -> `A` -> `B`. `A` get's a new `set` because there `cls.set == Artist.set`. But `B` will not get it's own `set` because `B.set == A.set != Artist.set`. Thus B.set will show only the properties of `A` not the additional ones of `B`.


We want to *always* define a new `set` unless the class definition itself contains a set (which none of our Artists does, but some third party might have implemented a dedicated set in their derived Artists and we don't want to break that). And I think the correct way to check this is checking `cls.__dict__`.

Reply:
> which none of our Artists does, but some third party might have implemented a dedicated set in their derived Artists and we don't want to break that

My point was that if thirdpartylib defines both A and B (in your example above) where A overrides `set` but B doesn't (and thus inherits `A.set`), I think that your current implementation will correctly leave A alone, but not B?

Reply:
Solved that via a flag on our autogenerated methods.

That said, I'm not 100% percent sure that this magic setter addition is what we want and downstream libraries expect. The alternative would be a class decorator, which we have to apply to all Artists; meaning adding this function would be opt-in.

Reply:
I think this is fine for now, AFAICT this will not really affect 3rd-parties now that you added the flag?